src/main/java/org/reaktivity/nukleus/socks/internal/stream/client/ConnectReplyStreamProcessor.java

```diff
@@ -0,0 +1,422 @@
+
+            // Can safely release the buffer
+            if (slotIndex != NO_SLOT)
+            {
+                context.bufferPool.release(slotIndex);
+                slotOffset = 0;
+                slotIndex = NO_SLOT;
+            }
+        }
+        else if (slotIndex == NO_SLOT)
+        {
+            if (NO_SLOT == (slotIndex = context.bufferPool.acquire(correlation.connectStreamId())))
+            {
+                doReset(connectReplyThrottle, connectReplyStreamId);
+                return;
+            }
+            MutableDirectBuffer acceptBuffer = context.bufferPool.buffer(slotIndex);
+            acceptBuffer.putBytes(0, buffer, offset, size);
+            slotOffset = size;
+        }
+    }
+
+    @State
+    private void afterConnectionResponse(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        switch (msgTypeId)
+        {
+        case DataFW.TYPE_ID:
+            final DataFW data = context.dataRO.wrap(buffer, index, index + length);
+            handleHighLevelData(data);
+            break;
+        case EndFW.TYPE_ID:
+            final EndFW end = context.endRO.wrap(buffer, index, index + length);
+            handleEnd(end);
+            break;
+        case AbortFW.TYPE_ID:
+            final AbortFW abort = context.abortRO.wrap(buffer, index, index + length);
+            handleAbort(abort);
+            break;
+        default:
+            doReset(connectReplyThrottle, connectReplyStreamId);
+            break;
+        }
+    }
+
+    private void handleHighLevelData(DataFW data)
+    {
+        OctetsFW payload = data.payload();
+        DataFW dataForwardFW = context.dataRW.wrap(context.writeBuffer, 0, context.writeBuffer.capacity())
+            .streamId(correlation.acceptReplyStreamId())
+            .payload(p -> p.set(
+                payload.buffer(),
+                payload.offset(),
+                payload.sizeof()))
+            .extension(e -> e.reset())
+            .build();
+        correlation.acceptReplyEndpoint()
+            .accept(
+                dataForwardFW.typeId(),
+                dataForwardFW.buffer(),
+                dataForwardFW.offset(),
+                dataForwardFW.sizeof());
+    }
+
+    private void handleEnd(EndFW end)
+    {
+        EndFW endForwardFW = context.endRW
+            .wrap(context.writeBuffer, 0, context.writeBuffer.capacity())
+            .streamId(correlation.acceptReplyStreamId())
+            .build();
+        correlation.acceptReplyEndpoint()
+            .accept(
+                endForwardFW.typeId(),
+                endForwardFW.buffer(),
+                endForwardFW.offset(),
+                endForwardFW.sizeof());
+    }
+
+    private void handleAbort(AbortFW abort)
+    {
+        doAbort(correlation.acceptReplyEndpoint(), correlation.acceptReplyStreamId());
+        correlation.acceptTransitionListener().transitionToAborted();
```

Comment:
It seems like the Listener is never updated, in which case the Listener abstraction is just another abstraction.  Seems like this should just be abort() and it should maybe be correlation.accept().abort() or correlation.acceptStream.handleConnectAbort()?  Why expose the state outside of the internal class?

Reply:
Agreed on the "another abstraction". The logic was "program to interfaces not implementations" and the idea was not to use the whole AccceptStreamProcessor in the ConnectReplyStreamProcessor only a small subset clearly defined. I guess I could change the type of the reference to the AcceptStreamProcessor and remove the interface. The intent was to isolate when the state changes from one side are triggered by the other side.

Reply:
This is an ok starting point.

It might be cleaner to use the same MessageConsumer state machine between the AcceptProcessor and the ConnectProcessor.  Then instead of telling it to abort, you pass an "Abort"/"Reset" message to the other side.  The state edges (messages) that each side then needs to implement should become pretty clear and easy to reason about.  This is the approach taken in HTTP cache.